src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,175 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/replace_symbolic.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+
+
+namespace LCompilers {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+
+class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisitor>
+{
+public:
+    ReplaceSymbolicVisitor(Allocator &al_) :
+    PassVisitor(al_, nullptr) {
+        pass_result.reserve(al, 1);
+    }
+
+    bool symbolic_replaces_with_CPtr_Module = false;
+    bool symbolic_replaces_with_CPtr_Function = false;
+
+    void visit_Module(const ASR::Module_t &x) {
+        SymbolTable* current_scope_copy = current_scope;
+        current_scope = x.m_symtab;
+        for (auto &a : x.m_symtab->get_scope()) {
+            this->visit_symbol(*a.second);
+            if(symbolic_replaces_with_CPtr_Module){
+                std::string new_name = current_scope->get_unique_name("basic_new_stack");
```

Comment:
```suggestion
                std::string new_name = "basic_new_stack";
```
`get_unique_name` checks the symtab if the `name` is declared. If it does, it gives the `unique_name`.
`get_unique_name` doesn't trigger the below if stmt.